app/Community/Components/UserProfileMeta.php

```diff
@@ -0,0 +1,227 @@
+            $ticketsResolvedForOthers += $ticketData['ResolvedCount'];
+        }
+
+        return compact(
+            'gameAuthoredAchievementsCount',
+            'openTickets',
+            'ticketsResolvedForOthers',
+            'totalAuthoredCodeNotes',
+            'totalAuthoredLeaderboards',
+        );
+    }
+
+    private function buildSocialStats(User $user): array
+    {
+        $userSetRequestInformation = getUserRequestsInformation($user);
+        $numForumPosts = $user->forumPosts()->count();
+
+        return compact(
+            'numForumPosts',
+            'userSetRequestInformation',
+        );
+    }
+
+    private function calculateAverageFinishedGames(array $userJoinedGamesAndAwards): string
+    {
+        $totalGames = count($userJoinedGamesAndAwards);
+        $finishedGames = 0;
+
+        // Iterate over each game to check if it is finished.
+        foreach ($userJoinedGamesAndAwards as $game) {
+            if (isset($game['HighestAwardKind']) && $game['HighestAwardKind'] !== null) {
+                $finishedGames++;
+            }
+        }
+
+        // Calculate the average percentage of finished games.
+        $averageFinishedGames = 0;
+        if ($totalGames > 0) {
+            $averageFinishedGames = ($finishedGames / $totalGames) * 100;
+        }
+
+        // Format and return the result to 2 decimal places.
+        return number_format($averageFinishedGames, 2, '.', '');
+    }
+
+    private function calculateAveragePointsPerWeek(string $userMemberSince, int $points = 0): int
+    {
+        $memberSince = Carbon::createFromFormat('Y-m-d H:i:s', $userMemberSince);
+        $now = Carbon::now();
+
+        $weeksOfMembership = $memberSince->diffInWeeks($now);
+
+        // Avoid division by zero.
+        if ($weeksOfMembership == 0) {
+            return $points;
+        }
+
+        $averagePointsPerWeek = $points / $weeksOfMembership;
+
+        return (int) round($averagePointsPerWeek);
+    }
+
+    private function calculateRecentPointsEarned(User $user, string $preferredMode = 'hardcore'): array
+    {
+        $thirtyDaysAgo = now()->subDays(30)->startOfDay();
+
+        $dateColumn = $preferredMode === 'hardcore' ? 'unlocked_hardcore_at' : 'unlocked_at';
+
+        $achievements = $user->playerAchievements()
+            ->with('achievement')
+            ->where($dateColumn, '>=', $thirtyDaysAgo)
+            ->get();
+
+        $pointsLast30Days = 0;
+        $pointsLast7Days = 0;
+
+        $now = now();
+
+        foreach ($achievements as $playerAchievement) {
+            $achievementDate = $playerAchievement->{$dateColumn};
+            $daysAgo = $now->diffInDays($achievementDate, false);
+
+            if ($daysAgo <= 30) {
+                $pointsLast30Days += $playerAchievement->achievement->points;
+            }
+            if ($daysAgo <= 7) {
+                $pointsLast7Days += $playerAchievement->achievement->points;
+            }
```

Comment:
Since the query only returns entries for the last 30 days, this could be simplified:
```
if ($daysAgo <= 7) {
    $pointsLast7Days += $playerAchievement->achievement->points;
} else {
    $pointsLast30Days += $playerAchievement->achievement->points;
}
```

Reply:
Yep, had some cruft leftover from something else I was trying in here. Simplified in latest.

Reply:
I misled you here. The else should be eliminated. Last 30 days should include the last 7 days.

Reply:
Fixed in latest.

Reply:
You misunderstood me. The `else` should be eliminated, not the entire `else` clause:
```
if ($daysAgo <= 7) {
    $pointsLast7Days += $playerAchievement->achievement->points;
} 

$pointsLast30Days += $playerAchievement->achievement->points;
```

Reply:
https://github.com/RetroAchievements/RAWeb/pull/2175/commits/c872060b51469a306fe1981d40f2bf386b721e93